File: 3434-find-the-number-of-distinct-colors-among-the-balls/3434-find-the-number-of-distinct-colors-among-the-balls.py
class Solution:
    def queryResults(self, limit: int, queries: List[List[int]]) -> List[int]:
        mpp1={}
        mpp2={}
        ans=[]
        for it in queries:
            if it[0] in mpp1:
                mpp2[mpp1[it[0]]]-=1
                if mpp2[mpp1[it[0]]]==0:
                    mpp2.pop(mpp1[it[0]])
            mpp1[it[0]]=it[1]
            if mpp1[it[0]] in mpp2 :
                mpp2[mpp1[it[0]]]+=1
            else:
                mpp2[mpp1[it[0]]]=1
            ans.append(len(mpp2))
        
        return ans

                
        

File: 3434-find-the-number-of-distinct-colors-among-the-balls/3434-find-the-number-of-distinct-colors-among-the-balls.cpp
class Solution {
public:
    vector<int> queryResults(int limit, vector<vector<int>>& queries) {
        map<int,int>mpp1;
        map<int,int>mpp2;
        vector<int>ans;
        for(auto it : queries){
           if(mpp1.find(it[0])!=mpp1.end()){
            mpp2[mpp1[it[0]]]--;
            if(mpp2[mpp1[it[0]]]==0) mpp2.erase(mpp1[it[0]]);
           }
           mpp1[it[0]]=it[1];
           mpp2[mpp1[it[0]]]++;
           ans.push_back(mpp2.size());

        }
        return ans;
    }
};